### PR TITLE
fix(propagation): adds support for propagation list.

### PR DESCRIPTION
**Pull request by jcchavezs**
## Description

This PR tries to address propagation issues being:

- Propagation is a list of supported formats instead of a single one.
- Propagation can be more than b3 or !b3
- Propagation is being set globaly rather than using a parameter b3-nob3 on instrumentations.

### CODE REVIEW

File: src/hypertrace/agent/config/__init__.py [hunk 1]
```diff
@@ -126,74 +130,75 @@ def __init__(self):  # pylint: disable=R0912,R0915
...
             self.config['data_capture']['body_max_size_bytes'] \
                 = int(os.environ['HT_DATA_CAPTURE_BODY_MAX_SIZE_BYTES'])
 
-        #Valid values are 'TRACECONTEXT' or 'B3'
-        if 'HT_PROPAGATION_FORMATS' in os.environ:
+        # Valid values are 'TRACECONTEXT' and/or 'B3'
+        if 'HT_PROPAGATION_FORMATS' in os.environ and len(os.environ['HT_PROPAGATION_FORMATS']) > 0:
             logger.debug("[env] Loaded HT_PROPAGATION_FORMATS from env")
-            self.config['propagation_formats'] = os.environ['HT_PROPAGATION_FORMATS']
+            self.config['propagation_formats'] = os.environ['HT_PROPAGATION_FORMATS'].split(
```
Comment (jcchavezs) [inline src/hypertrace/agent/config/__init__.py:200]: This is important change @rcbjBlueMars @nsahai

File: src/hypertrace/agent/config/__init__.py [hunk 2]
```diff
@@ -263,13 +268,17 @@ def __init__(self):  # pylint: disable=R0912,R0915
         self.agent_config.service_name = self.config['service_name']
         self.agent_config.reporting = reporting
         self.agent_config.data_capture = data_capture
-        if self.config['propagation_formats']  == 'TRACECONTEXT':
-            self.agent_config.propagation_formats = config_pb2.PropagationFormat.TRACECONTEXT
-        elif self.config['propagation_formats'] == 'B3':
-            self.agent_config.propagation_formats = config_pb2.PropagationFormat.B3
+        if 'TRACECONTEXT' in self.config['propagation_formats']:
+            self.agent_config.propagation_formats.add(
```
Comment (jcchavezs) [inline src/hypertrace/agent/config/__init__.py:272]: This part I could not make it work. This is supposed to be a list but I could not make it work like that and could not decypher the methods for this object. Would you help me with that @rcbjBlueMars ?

File: src/hypertrace/agent/config/default.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 DEFAULT_AGENT_CONFIG = {
     '_use_console_span_exporter': False,
     'enabled': True,
-    'propagation_formats':'TRACECONTEXT',
+    'propagation_formats': ['TRACECONTEXT'],
```
Comment (jcchavezs) [inline src/hypertrace/agent/config/default.py:5]: This is key @rcbjBlueMars @nsahai

File: src/hypertrace/agent/init/__init__.py [hunk 4]
```diff
@@ -57,49 +54,63 @@ def __init__(self, agent_config: AgentConfig, init_console_only: bool = False):
...
+                "service.name": self._config.agent_config.service_name,
+                "service.instance.id": os.getpid(),
+            })
+        )
+
+        trace.set_tracer_provider(self._tracer_provider)
+
+    def init_propagation(self) -> None:
+        print('------')
+        print(type(self._config.agent_config.propagation_formats))
```
Comment (jcchavezs) [inline src/hypertrace/agent/init/__init__.py:76]: Here I tried to understand the value for it with no success yet.

File: src/hypertrace/agent/config/__init__.py (hunk 2 above)
Comment (rcbjBlueMars) [reply src/hypertrace/agent/config/__init__.py:272]: I've gotten up to this PR for merging all outstanding changes. I'm going to merge in the latest main branch, fix the parsing logic, and  update all tests to correctly use an array of strings.

File: src/hypertrace/agent/config/default.py (hunk 3 above)
Comment (rcbjBlueMars) [reply src/hypertrace/agent/config/default.py:5]: Fixed.

File: src/hypertrace/agent/init/__init__.py (hunk 4 above)
Comment (rcbjBlueMars) [reply src/hypertrace/agent/init/__init__.py]: Fixed.

**Pull request merged**